### PR TITLE
STEEl-92 - network gateway resource fixed up, tests written and docum…

### PR DESCRIPTION
**Pull request by mastamark**
1200px-gw2k_2004_logo svg
…ented, datasource created and documented, and a few typos

### CODE REVIEW

File: rightscale/data_source_network_gateway.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+							Type:        schema.TypeString,
+							Description: "filter by network href",
+							Optional:    true,
+							ForceNew:    true,
+						},
+					},
+				},
+			},
+
+			// Read-only fields
```
Comment (raphael) [inline rightscale/data_source_network_gateway.go]: description is not computed?

File: website/docs/d/cm_network_gateway.markdown [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+* `name` - Name of the network gateway.
+
+* `resource_uid` - Network gateway resource_uid as reported by cm platform.
+
+* `type` - Type of network gateway.  Options are "internet" or "vpc."
+
+* `state` - State of the network gateway as reported by cm platform.  ("available" means attached to a network)
+
+* `description` - The description of the network.
```
Comment (raphael) [inline website/docs/d/cm_network_gateway.markdown]: gateway

File: website/docs/d/cm_network_gateway.markdown [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+The following attributes are exported:
+
+* `name` - Name of the network gateway.
+
+* `resource_uid` - Network gateway resource_uid as reported by cm platform.
+
+* `type` - Type of network gateway.  Options are "internet" or "vpc."
+
+* `state` - State of the network gateway as reported by cm platform.  ("available" means attached to a network)
```
Comment (raphael) [inline website/docs/d/cm_network_gateway.markdown]: not sure we need "as reported by cm platform"
would be good to enumerate possible values

File: website/docs/d/cm_network_gateway.markdown [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+  * `network_href` - (Optional) Network HREF network gateways are attached to.
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `name` - Name of the network gateway.
+
+* `resource_uid` - Network gateway resource_uid as reported by cm platform.
```
Comment (raphael) [inline website/docs/d/cm_network_gateway.markdown]: not sure we need "as reported by cm platform" (more confusing than helpful to non-RS people)
could explain that resource_uid is underlying native cloud id

File: website/docs/r/cm_network_gateway.markdown [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+* `description` - (Optional) Network gateway description.
+
+* `network_href` - (Optional) HREF of network you want to attach the network gateway to.
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `href` - Href of the network gateway.
```
Comment (raphael) [inline website/docs/r/cm_network_gateway.markdown]: HREF? should probably be consistent
Also note that it should really be `href` everywhere. It's not an acronym.

File: website/docs/r/cm_network_gateway.markdown [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+* `created_at` - Date the network gateway was created at.
+
+* `updated_at` - Date the network gateway was updated at.
+
+* `state` - State of the network gateway as reported by cm platform.  ("available" means attached to a network)
```
Comment (raphael) [inline website/docs/r/cm_network_gateway.markdown]: same comments as for datasource for `state` and `resource_uid`.

File: website/docs/d/cm_network_gateway.markdown (hunk 3 above)
Comment (mastamark) [reply website/docs/d/cm_network_gateway.markdown]: I'll remove the 'as reported by cm platform' but I have no idea what the actual possible states ARE here - http://reference.rightscale.com/api1.5/media_types/MediaTypeNetworkGateway.html

I got the "available" means attached to a network from a tech tool in the UI networking app.

**Pull request merged**